### DISCUSSION TITLE
Feature request: Environment variable containing GitHub username

### DISCUSSION THREAD

**Discussion by dcamper** [community/community#3994] [Codespaces] [answered]
It would be helpful in some situations to have access to the current user's GitHub username during VM setup.  An environment variable seems to be a good place for that.

In our case, we would use this to customize a VS Code launch.json document that is used to by an extension to access an outside service.

**Comment by bailey** [comment, marked as answer]
Hey there @dcamper ! Thanks for this feedback, that actually sounds like a great use of our [secrets functionality](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces), as it's an easy way to store environment variables. Have you given that a try at all?

**Reply by dcamper** [reply]
How would the value of the secret be populated, though?  I'm looking for the GitHub username of the person launching the Codespace, not the user creating the Codespace.

**Reply by 2percentsilk** [reply]
The user who creates the codespace will be the only person who can launch/access that codespace so you could add the username for your own codespaces and then access that once you're within the codespace!

**Reply by dcamper** [reply]
@2percentsilk I understand what you're saying, but I don't think that addresses my request.  I would like to be able to customize the codespace -- which means during setup of the container, not after -- using the username of the user that launched the codespace, all without requiring the user to do anything special other than being logged in to Github.

**Reply by tanmayeekamath** [reply]
@dcamper sounds like you want to create a Codespace on behalf of another user? 
Currently that is not supported, however it is fairly straightforward for a user to request a Codespace from their repo.  

I would love to learn more on the problem you are trying to address with this request, can you help share context around that? :)

**Reply by dcamper** [reply]
The context here involves a client/server architecture.  A running codespace instance would represent a client.  The server side of things would be configured via a settings file within the repo.  The goal is to customize a different part of that settings file with the current user's github username, so when the client (running in the new codespace) connects with the server, that username is already known.

As a high-level example, this kind of setup could be used in a classroom environment.  All students would start a codespace given to them by the instructor via a github URL.  The repo would contain all settings and have class materials (including code to run).  The VM that is started acts as the students' execution environment, and the idea is that each environment is _slightly_ modified with the student's username so that the instructor can tell their work apart on the server side.  To make this as easy as possible, we'd like to automatically insert the github username into the configuration file during the initial codespace creation phase.

**Reply by tanmayeekamath** [reply]
Thank you so much for the detailed context there, @dcamper! I will capture this scenario and request in our backlog. :)
